kazoo/protocol/connection.py

```diff
@@ -808,23 +806,11 @@ def _authenticate_with_sasl(self, host, timeout):
             try:
                 response = sasl_cli.process(challenge=challenge)
             except puresasl.SASLError as err:
-                six.reraise(
-                    SASLException,
-                    SASLException("library error: %s" % err),
-                    sys.exc_info()[2],
-                )
+                raise SASLException("library error: %s" % err) from err
```

Comment:
Do we still need the `: %s" % err` in these three?

Would `raise SASLException("library error") from err` work?

Reply:
```
>>> def throw():
...     raise KeyError("Message 1")
...
>>> def throw_through():
...     try:
...         throw()
...     except KeyError as err:
...         raise Exception("Message 2") from err
...
>>> throw_through()
Traceback (most recent call last):
  File "<stdin>", line 3, in throw_through
  File "<stdin>", line 2, in throw
KeyError: 'Message 1'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in throw_through
Exception: Message 2
```

This is what this would produce. Happy to drop it, just want to make sure we're happy with the "change" in exception messaging

Reply:
I think this is OK.  Having the stringification of parent exception in the message does not provide any value.

We only expose "library", "protocol" and "unknown" messages at this point anyway. I think this is good 👍